src/binary-reader-ir.cc

```diff
@@ -1447,6 +1517,38 @@ Result BinaryReaderIR::OnLocalNameLocalCount(Index index, Index count) {
   return Result::Ok;
 }
 
+Result BinaryReaderIR::BeginCodeMetadataSection(std::string_view name,
+                                                  Offset size) {
+  current_metadata_name_ = name;
+  return Result::Ok;
+}
+
+Result BinaryReaderIR::OnCodeMetadataFuncCount(Index count) {
+  return Result::Ok;
+}
+
+Result BinaryReaderIR::OnCodeMetadataCount(Index function_index,
+                                             Index count) {
+  code_metadata_queue_.push_func(module_->funcs[function_index]);
+  return Result::Ok;
+}
+
+Result BinaryReaderIR::OnCodeMetadata(Offset offset,
+                                        const void* data,
+                                        Address size) {
+  std::vector<uint8_t> data_(static_cast<const uint8_t*>(data),
+                             static_cast<const uint8_t*>(data) + size);
+  auto meta = MakeUnique<CodeMetadataExpr>(current_metadata_name_,
+                                            std::move(data_));
+  meta->loc.offset = offset;
+  code_metadata_queue_.push_metadata(std::move(meta));
+  return Result::Ok;
+}
+
+Result BinaryReaderIR::EndCodeMetadataSection() {
+  return Result::Ok;
```

Comment:
Maybe just remove this ?

Reply:
ok. I added it for completeness but it is not necessary, at least for now